rewrite-java/src/main/java/org/openrewrite/java/ChangePackage.java

```diff
@@ -390,13 +390,12 @@ private boolean isTargetRecursivePackageName(String packageName) {
     private static class ReferenceChangePackageVisitor extends TreeVisitor<Tree, ExecutionContext> {
         Map<Tree, Reference> matches;
         Reference.Renamer renamer;
-        String newPackageName;
 
         @Override
         public @Nullable Tree preVisit(@Nullable Tree tree, ExecutionContext ctx) {
```

Comment:
I think we should change this to be a `postVisit()` method instead. That way the cursor available to the `Reference` should still be valid while we call the `rename()` method.